### ISSUE TITLE
[CAL-2341] Can't invite team members with username

### ISSUE THREAD

**Issue by ghost** [calcom/cal.com#10726]
### Issue Summary

Can't invite team members with username even though the input label is 'Email or Username'. On submitting the form with username the error is 'Invite failed because "username" is not a valid email address'.

### Steps to Reproduce

1. Try to add a member in your team with their username

### Screenshot

![](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/479fd7b9-b9b4-4cbf-98af-b1da194dcfdf/25bca784-c5ab-4a50-83c9-7fecb07e888d?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2U4NmJmOTU3LWQ4MmYtNDY1ZS1iMjA1LTEzNTU1OWY0YjYyMy80NzlmZDdiOS1iOWI0LTRjYmYtOThhZi1iMWRhMTk0ZGNmZGYvMjViY2E3ODQtYzVhYi00YTUwLTgzYzktN2ZlY2IwN2U4ODhkIiwiaWF0IjoxNjkxODc4ODA0LCJleHAiOjE2OTE5NjUyMDR9.8MIJbbAyyw079ySAzdKsJO_Bowo0x3Z9hOUF2QZFbBE)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2341](https://linear.app/calcom/issue/CAL-2341/cant-invite-team-members-with-username)</sub>

**Comment by RohanMishra315** [comment]
Hey , @vineetpradhan-dev  I solve these issue on my local server pls assign me. I will provide you the code as well if you want.
So assign me pls.

**Comment by rijuoddusual1999** [comment]
Hii @vineetpradhan-dev  @RohanMishra315 , I am new in this site.  I have followed the setup guidance but after **yarn dx**. I am getting some error which I will attach below and after running **yarn dev** I am not able to pull the host on http://localhost:3000


The issue that I got:
command (C:\Users\pilot\OneDrive\Desktop\cal.com #10726\cal.com\packages\prisma) yarn run db-up exited (1)

 Tasks:    1 successful, 2 total
Cached:    0 cached, 2 total
  Time:    3.601s
Failed:    @calcom/prisma#db-up

 ERROR  run failed: command  exited (1)

**Comment by InderdeepBajwa** [comment]
The issue is arising from: https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts#L79

The check for `{ username: usernameOrEmail, organizationId: orgId }` containing `organizationId: orgId` is causing this issue. As the username that is already not in the organization will not return any result.

Was this feature meant to invite any user across Cal.com and not just within a defined organization?

If yes, the fix is to change `{ username: usernameOrEmail, organizationId: orgId }` to `{ username: usernameOrEmail }`

**Comment by RohitSaini15** [comment]
> Hii @vineetpradhan-dev @RohanMishra315 , I am new in this site. I have followed the setup guidance but after **yarn dx** I am getting some error which I will attach below and after running **yarn dev** I am not able to pull the host on http://localhost:3000
> 
> The issue that I got: command (C:\Users\pilot\OneDrive\Desktop\cal.com #10726\cal.com\packages\prisma) yarn run db-up exited (1)
> 
> Tasks: 1 successful, 2 total Cached: 0 cached, 2 total Time: 3.601s Failed: @calcom/prisma#db-up
> 
> ERROR run failed: command exited (1)

Docker and Docker-compose both should be installed before running the command.

**Comment by RohitSaini15** [comment]
> Hii @vineetpradhan-dev @RohanMishra315 , I am new in this site. I have followed the setup guidance but after **yarn dx** I am getting some error which I will attach below and after running **yarn dev** I am not able to pull the host on http://localhost:3000
> 
> The issue that I got: command (C:\Users\pilot\OneDrive\Desktop\cal.com #10726\cal.com\packages\prisma) yarn run db-up exited (1)
> 
> Tasks: 1 successful, 2 total Cached: 0 cached, 2 total Time: 3.601s Failed: @calcom/prisma#db-up
> 
> ERROR run failed: command exited (1)

Docker and Docker-compose both should be installed before running the command.

**Comment by vineetpradhan-dev** [comment]
@InderdeepBajwa I think the "checkInputEmailIsValid" function will still going to create confusion. If someone misspells username and it doesn't find invitee, the error would be "invalid email".

**Comment by InderdeepBajwa** [comment]
That is correct. I see where the confusion is arising, especially with the way `if (!invitee)` is coded.

**Comment by RohanMishra315** [comment]
hey @InderdeepBajwa make sure docker is installed and running on desktop when you setup the source code.

**Comment by prane-eth** [comment]
@vineetpradhan-dev, I fixed it locally. Can you assign the issue to me?

**Comment by prane-eth** [comment]
@vineetpradhan-dev, could you please review #10730?

**Comment by vineetpradhan-dev** [comment]
@prane-eth , I don't know how to request review here. And can you please test if you can add user who are already in the team.

**Comment by rijuoddusual1999** [comment]
@RohitSaini15  I have Docker desktop installed already but during yarn dx it is failing in something **prisma packages**. In docker I can see the email log instance as well.

**Comment by RohitSaini15** [comment]
@rijuoddusual1999 for windows, copy the content of .env file in packages/prisma/.env then run yarn dx

**Comment by rijuoddusual1999** [comment]
it worked, thanks a lot @RohitSaini15.

**Comment by rijuoddusual1999** [comment]
My initial guess is to be a valid email address there must be a **@** but a username most probably will not have it.  So I fixed it my using a **AND** operator which will look in both cases and it worked on my local machine but it is failing unit tests. 


**export function checkInputEmailIsValid(input: string) {
  if (!isEmail(input) && isEmail(input)) {
    throw new TRPCError({
      code: "BAD_REQUEST",
      message: `Invite failed because ${input} is neither a valid email address nor a valid username`,
    });
  }
}**



I changed this particular function in **packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts**

If there is anything I can do please let me know.

**Comment by prane-eth** [comment]
@rijuoddusual1999, `!isEmail(input) && isEmail(input)` will always return false irrespective of the input. I fixed it in a better way here:
https://github.com/calcom/cal.com/pull/10730/files

**Comment by rijuoddusual1999** [comment]
@prane-eth  it worked in my local machine, I was able to add both user name and email address. 

![vscode snippet](https://github.com/calcom/cal.com/assets/113231164/69f103f7-b992-4fca-96c6-81a987b56db6)
![website snippet](https://github.com/calcom/cal.com/assets/113231164/01cb48b1-2662-4225-8b33-1f9be4c1120e)

I completely agree with you it should been an OR operator but I used OR for my first case and it was not taking emails. Surprisingly the AND operator worked . 

I was not able to come up with the function you carried as I was going through the code base in hurry, but I liked how you solved it completely from the root.
